it/uk/gov/hmrc/agentaccesscontrol/support/BaseISpec.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentaccesscontrol.support
+
+import com.github.tomakehurst.wiremock.client.WireMock._
+import org.scalatestplus.play.OneServerPerSuite
+import uk.gov.hmrc.agentaccesscontrol.StartAndStopWireMock
+import uk.gov.hmrc.play.http.HeaderCarrier
+import uk.gov.hmrc.play.test.UnitSpec
+
+abstract class BaseISpec extends UnitSpec
+    with StartAndStopWireMock
+    with StubUtils
+    with OneServerPerSuite {
+  implicit val hc = HeaderCarrier()
+}
+
+trait StubUtils {
+  me: StartAndStopWireMock =>
+
+  class PreconditionBuilder {
+    def agentAdmin(agentCode: String, oid: String = "556737e15500005500eaf68e") = {
+      new AgentAdmin(agentCode, oid)
+    }
+  }
+
+  def given() = {
+    new PreconditionBuilder()
+  }
+
+  class AgentAdmin(agentCode: String, oid: String) {
+    def andGettingEnrolmentsFailsWith500(): AgentAdmin = {
+      stubFor(get(urlPathEqualTo(s"/auth/oid/$oid/enrolments")).willReturn(aResponse().withStatus(500)))
+      this
+    }
+
+    def andHasNoSaAgentReference(): AgentAdmin = {
+      stubFor(get(urlPathEqualTo(s"/auth/oid/$oid/enrolments")).willReturn(aResponse().withStatus(200).withBody(
+        s"""
+           |[{"key":"IR-PAYE-AGENT","identifiers":[{"key":"IrAgentReference","value":"HZ1234"}],"state":"Activated"},
+           | {"key":"HMRC-AGENT-AGENT","identifiers":[{"key":"AgentRefNumber","value":"JARN1234567"}],"state":"Activated"}]
+         """.stripMargin
+      )))
+      this
+    }
+
+    def andHasSaAgentReference(saAgentReference: String): AgentAdmin = {
+      stubFor(get(urlPathEqualTo(s"/auth/oid/$oid/enrolments")).willReturn(aResponse().withStatus(200).withBody(
+        s"""
+           |[{"key":"IR-PAYE-AGENT","identifiers":[{"key":"IrAgentReference","value":"HZ1234"}],"state":"Activated"},
+           | {"key":"HMRC-AGENT-AGENT","identifiers":[{"key":"AgentRefNumber","value":"JARN1234567"}],"state":"Activated"},
+           | {"key":"IR-SA-AGENT","identifiers":[{"key":"IRAgentReference","value":"$saAgentReference"}],"state":"Activated"}]
```

Comment:
I think the code is missing a filter on key = "IRAgentReference".

Adding another identifier by changing this line to the following provokes a test failure:
``` {"key":"IR-SA-AGENT","identifiers":[{"key":"AnotherIdentifier", "value": "not the IR Agent Reference"}, {"key":"IRAgentReference","value":"$saAgentReference"}],"state":"Activated"}]```